File: Data Structures & Algorithms/valid-binary-search-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    bool isValidBST(TreeNode* root) {
        return c(root,INT_MIN,INT_MAX   );
    }
    bool c(TreeNode* root,int l,int h){
        if(!root)return true;
        if(l>=root->val or h<=root->val)return false;
        return c(root->left,l,root->val) and c(root->right,root->val,h);
        // return false;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n=len(nums)
        left=[0]*n
        right=[0]*n
        res=[0]*n
        left[0]=right[n-1]=1
        i=1
        while i<n:
            left[i]=left[i-1]*nums[i-1]
            i+=1
        # i=n-2
        for x in range (n-2,-1,-1):
            i=x
            right[i]=right[i+1]*nums[i+1]

        for i in range (0,n):
            res[i]=left[i]*right[i]
        return res